### PR TITLE
Amazon equiv mongo title index

### PR DESCRIPTION
**Pull request by tristanbeardmore-gray**
A mongo implementation to store an index of all Amazon content sharing the same title.

### CODE REVIEW

File: src/main/java/org/atlasapi/remotesite/amazon/AmazonPersistenceModule.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+@Configuration
+public class AmazonPersistenceModule {
+
+    @Autowired private Mongo mongo;
+
+    private final Parameter processingConfig = Configurer.get("processing.config");
```
Comment (dragospitica) [inline src/main/java/org/atlasapi/remotesite/amazon/AmazonPersistenceModule.java]: What's the use of this field in this class?

File: src/main/java/org/atlasapi/remotesite/amazon/indexer/AmazonTitleIndexEntry.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+public class AmazonTitleIndexEntry {
+    private final String title;
+    private final Set<String> uris;
+    private DateTime lastUpdated;
+
+    public AmazonTitleIndexEntry(
+            String title,
+            @Nullable Collection<String> uris,
+            @Nullable DateTime lastUpdated
```
Comment (dragospitica) [inline src/main/java/org/atlasapi/remotesite/amazon/indexer/AmazonTitleIndexEntry.java]: Do we need a constructor with lastUpdated if we set it anyway before writing to DB?

Comment (dragospitica) [reply src/main/java/org/atlasapi/remotesite/amazon/indexer/AmazonTitleIndexEntry.java]: Hmm, yes we need to deserialize the DB entry 👍

File: src/main/java/org/atlasapi/persistence/content/mongo/MongoContentLister.java [hunk 3]
```diff
@@ -107,9 +105,10 @@ public DBObject queryForCategory(ContentCategory category) {
                     public DBCursor cursorFor(ContentCategory category) {
                         return contentTables.collectionFor(category)
                                 .find(queryForCategory(category))
-                                .batchSize(100)
+                                .batchSize(1)
                                 .sort(new MongoSortBuilder().ascending("publisher").ascending(MongoConstants.ID).build())
-                                .addOption(Bytes.QUERYOPTION_NOTIMEOUT);
+                                .addOption(Bytes.QUERYOPTION_NOTIMEOUT)
+                                .noCursorTimeout(true);
```
Comment (EmiEnache) [inline src/main/java/org/atlasapi/persistence/content/mongo/MongoContentLister.java]: revert this, as it's been fixed by Emi's changes for running equiv on full publisher

Comment (EmiEnache) [reply src/main/java/org/atlasapi/persistence/content/mongo/MongoContentLister.java]: (and everything else in this class)

Comment (EmiEnache) [inline src/main/java/org/atlasapi/persistence/content/mongo/MongoContentLister.java]: TODO: undo changes in this class.

**Pull request merged**